akka-stream-typed/src/main/scala/akka/stream/typed/javadsl/ActorSource.scala

```diff
@@ -60,4 +60,31 @@ object ActorSource {
         overflowStrategy)
       .asJava
   }
+
+  /**
+   * Creates a `Source` that is materialized as an [[akka.actor.ActorRef]].
+   * Messages sent to this actor will be emitted to the stream if there is demand from downstream,
+   * and a new message will only be accepted after the previous messages has been consumed and acknowledged back.
+   * The stream will complete with failure if a message is sent before the acknowledgement has been replied back.
+   *
+   * The stream can be completed with failure by sending a message that is matched by `failureMatcher`. The extracted
+   * [[Throwable]] will be used to fail the stream. In case the Actor is still draining its internal buffer (after having received
+   * a message matched by `completionMatcher`) before signaling completion and it receives a message matched by `failureMatcher`,
+   * the failure will be signaled downstream immediately (instead of the completion signal).
+   *
+   * The actor will be stopped when the stream is completed, failed or canceled from downstream,
+   * i.e. you can watch it to get notified when that happens.
+   */
+  def actorRefWithAck[T, Ack](
+      ackTo: ActorRef[Ack],
+      ackMessage: Ack,
+      completionMatcher: PartialFunction[T, CompletionStrategy],
+      failureMatcher: PartialFunction[T, Throwable]): Source[T, ActorRef[T]] =
+    akka.stream.typed.scaladsl.ActorSource
+      .actorRefWithAck[T, Ack](
+        ackTo,
+        ackMessage,
+        completionMatcher.asInstanceOf[PartialFunction[Any, CompletionStrategy]],
+        failureMatcher.asInstanceOf[PartialFunction[Any, Throwable]])
```

Comment:
in `javadsl` we shouldn't use `PartialFunction`. Could it be 
```
java.util.Function[T, Optional[CompletionStrategy]]
java.util.Function[T, Optional[Throwable]]
```
?

Reply:
I've just followed the `actorRef` java api above which also uses `PartialFunction` (The sink java api use functions although not with optional).
I could switch the `actorRefWithAck` but what about `actorRef`?

Reply:
hm, that must have been an oversight. let's change that in a follow up PR, I created issue https://github.com/akka/akka/issues/26970